library/std/src/sys/pal/hermit/os.rs

```diff
@@ -154,30 +154,29 @@ impl Iterator for Env {
 /// Returns a vector of (variable, value) byte-vector pairs for all the
 /// environment variables of the current process.
 pub fn env() -> Env {
-    unsafe {
-        let guard = ENV.as_ref().unwrap().lock().unwrap();
-        let mut result = Vec::new();
+    let guard = ENV.lock().unwrap();
+    let env = guard.as_ref().unwrap();
+    let mut result = Vec::new();
 
-        for (key, value) in guard.iter() {
-            result.push((key.clone(), value.clone()));
-        }
-
-        return Env { iter: result.into_iter() };
+    for (key, value) in env {
+        result.push((key.clone(), value.clone()));
     }
```

Comment:
I'd use iterator adapters here to save some reallocations:
```suggestion
    let result = env.iter().map(|(key, value)| (key.clone(), value.clone())).collect::<Vec<_>>();
```